src/test/java/org/opensearch/neuralsearch/plugin/query/NeuralQueryIT.java

```diff
@@ -0,0 +1,269 @@
+            TEST_QUERY_TEXT,
+            modelId.get(),
+            1,
+            null
+        );
+
+        Map<String, Object> searchResponseAsMap = search(TEST_BASIC_INDEX_NAME, matchAllQueryBuilder, rescoreNeuralQueryBuilder, 1);
+        Map<String, Object> firstInnerHit = getFirstInnerHit(searchResponseAsMap);
+
+        assertEquals(firstInnerHit.get("_id"), "1");
+        float expectedScore = computeExpectedScore(modelId.get(), testVector, TEST_SPACE_TYPE);
+        assertEquals(expectedScore, objectToFloat(firstInnerHit.get("_score")), 0.0);
+    }
+
+    @SneakyThrows
+    public void testBooleanQuery() {
+        maybeInitializeIndex(TEST_MULTI_FIELD_INDEX_NAME);
+        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+
+        NeuralQueryBuilder neuralQueryBuilder1 = new NeuralQueryBuilder(
+            TEST_KNN_VECTOR_FIELD_NAME_1,
+            TEST_QUERY_TEXT,
+            modelId.get(),
+            1,
+            null
+        );
+        NeuralQueryBuilder neuralQueryBuilder2 = new NeuralQueryBuilder(
+            TEST_KNN_VECTOR_FIELD_NAME_2,
+            TEST_QUERY_TEXT,
+            modelId.get(),
+            1,
+            null
+        );
+
+        boolQueryBuilder.should(neuralQueryBuilder1).should(neuralQueryBuilder2);
+
+        Map<String, Object> searchResponseAsMap = search(TEST_MULTI_FIELD_INDEX_NAME, boolQueryBuilder, 1);
+        Map<String, Object> firstInnerHit = getFirstInnerHit(searchResponseAsMap);
+
+        assertEquals(firstInnerHit.get("_id"), "1");
+        float expectedScore = 2 * computeExpectedScore(modelId.get(), testVector, TEST_SPACE_TYPE);
+        assertEquals(expectedScore, objectToFloat(firstInnerHit.get("_score")), 0.0);
+    }
+
+    @SneakyThrows
+    public void testNestedQuery() {
+        maybeInitializeIndex(TEST_NESTED_INDEX_NAME);
+
+        NeuralQueryBuilder neuralQueryBuilder = new NeuralQueryBuilder(
+            TEST_KNN_VECTOR_FIELD_NAME_NESTED,
+            TEST_QUERY_TEXT,
+            modelId.get(),
+            1,
+            null
+        );
+
+        Map<String, Object> searchResponseAsMap = search(TEST_NESTED_INDEX_NAME, neuralQueryBuilder, 1);
+        Map<String, Object> firstInnerHit = getFirstInnerHit(searchResponseAsMap);
+
+        assertEquals(firstInnerHit.get("_id"), "1");
+        float expectedScore = computeExpectedScore(modelId.get(), testVector, TEST_SPACE_TYPE);
+        assertEquals(expectedScore, objectToFloat(firstInnerHit.get("_score")), 0.0);
+    }
+
+    @SuppressWarnings("unchecked")
+    protected Map<String, Object> getFirstInnerHit(Map<String, Object> searchResponseAsMap) {
+        Map<String, Object> hits1map = (Map<String, Object>) searchResponseAsMap.get("hits");
+        List<Object> hits2List = (List<Object>) hits1map.get("hits");
+        assertTrue(hits2List.size() > 0);
+        return (Map<String, Object>) hits2List.get(0);
+    }
+
+    @SneakyThrows
+    private String prepareModel() {
```

Comment:
should we move this to BaseNeuralSearchIT class?

Reply:
sure